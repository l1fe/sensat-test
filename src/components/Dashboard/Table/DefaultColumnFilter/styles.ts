import styled from 'styled-components';

export const Container = styled.div`
  width: 100%;
  padding: 0 4px;
  box-sizing: border-box;
`;

export const Input = styled.input`
  width: 100%;
  box-sizing: border-box;
`;
