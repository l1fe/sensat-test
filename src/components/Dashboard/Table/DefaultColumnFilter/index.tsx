import React, { useState } from 'react';
import debounce from 'lodash.debounce';
import { Container, Input } from './styles';

const DefaultColumnFilter: React.FC<{ column: any }> = ({
  column: { filterValue, preFilteredRows, setFilter },
}) => {
  const [value, setValue] = useState(filterValue);
  const count = preFilteredRows.length;

  const onChange = debounce((val: string | undefined) => {
    setFilter(val || undefined);
  }, 200);

  return (
    <Container>
      <Input
        value={value || ''}
        onChange={(e: React.ChangeEvent<HTMLInputElement>) => {
          setValue(e.target.value);
          onChange(e.target.value);
        }}
        placeholder={`Search ${count} records...`}
      />
    </Container>
  );
};

export default DefaultColumnFilter;
