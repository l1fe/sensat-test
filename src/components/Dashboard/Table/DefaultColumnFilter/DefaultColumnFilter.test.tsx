import renderer from 'react-test-renderer';
import DefaultColumnFilter from './';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <DefaultColumnFilter
        column={{
          filterValue: 'filter',
          preFilteredRows: [],
          setFilter: () => {},
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly - some prefiltered rows', () => {
  const tree = renderer
    .create(
      <DefaultColumnFilter
        column={{
          filterValue: 'f',
          preFilteredRows: [{ a: 1 }, { b: 2 }],
          setFilter: () => {},
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
