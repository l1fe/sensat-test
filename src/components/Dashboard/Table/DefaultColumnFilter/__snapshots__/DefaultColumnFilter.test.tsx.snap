// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - some prefiltered rows 1`] = `
<div
  className="sc-bdfBQB gZhhqC"
>
  <input
    className="sc-gsTEea eQcJzg"
    onChange={[Function]}
    placeholder="Search 2 records..."
    value="f"
  />
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="sc-bdfBQB gZhhqC"
>
  <input
    className="sc-gsTEea eQcJzg"
    onChange={[Function]}
    placeholder="Search 0 records..."
    value="filter"
  />
</div>
`;
